BTA-1907 Update API documentation to contain the new fields and rules (#6)
transferzero/api-documentation: transaction-flow.md

  "birth_date": "1970-01-01",
  "documents": [ ],
  "ip": "127.0.0.1",

>   "identification_number": "1234",

Reviewer: Can we also make this have more characters so it's more realistic? eg. AB123456

  * If you already do KYC on your system, then please contact us and we can waive this requirement from you. In this case you should send us an empty list of documents: `"documents": [ ]` in the request. All of the senders you create in the system will be immediately set to the `approved` state and you won't need to wait for them to get approved.
  * If when creating senders or transactions you get the following error in the response: `"errors":{"documents":[{"error":"blank"}]}` it means that KYC requirements are not yet waived for your account. If we already approved your KYC process and so they should be, then please contact us so we can rectify the issue and update your account accordingly.
  * In case you don't do KYC on your site, then you will need to send us documents that we can use to verify the sender's identity, for more details on this please see the [API reference documentation](https://api.transferzero.com/documentation#documents).

> * Identification number: Identification number of document used

Reviewer 2: Since there's a minimum 4-character limit to this, if provided, do we need to mention that here? (It's highly unlikely someone's going to have a 3-digit ID number, but more clarity in documentation seems better to me than the alternative.)

Author: Yeah I also thought it's unlikely some's going to have a 3-digit ID number but like you said better being as more specific, it can results in a less query/problem with our customers. I will add it now.

> * Identification number: Identification number of document used with a 4-character minimum length.

Reviewer: Don't think we need to specify the character validation. If it's less than 4 characters it's already wrong anyway.
```suggestion
* Identification number: Identification number of the document used.
```

Author: As per Slack messages the length of the identification number will be left in.

* Identification type:
  - `DL`: Driving License
  - `PP`: International Passport
  - `ID`: National ID
  - `OT`: Other
* There are three different sets of fields:
  - 'identification_number' and 'identification_type'

>   - 'city', 'street', 'address_description', 'postal_code'

Reviewer: address_description isn't mandatory anymore so we can remove it
```suggestion
  - 'city', 'street', 'postal_code'
```

  3. 'birth_date'

>   The sender only needs to contain one of the set of fields from above.

Reviewer 3: This should come as a separate paragraph. Please check the rendered output (rich diff button aboce in github) as now it just moves this text as part of the above

> * The sender only needs to contain one of the set of fields from above.

Reviewer 3: We need to add more details on how WTR2 works. E.g. this is only true if the WTR2 mode is enabled for the user

Reviewer 4: Here is the communication that we are looking to send out :
https://docs.google.com/document/d/1VlXSiWCcfGSbr6CPiRWNUai28WQa-hpMtHmLiVVrqe8/edit?usp=sharing

  * If you already do KYC on your system, then please contact us and we can waive this requirement from you. In this case you should send us an empty list of documents: `"documents": [ ]` in the request. All of the senders you create in the system will be immediately set to the `approved` state and you won't need to wait for them to get approved.
  * If when creating senders or transactions you get the following error in the response: `"errors":{"documents":[{"error":"blank"}]}` it means that KYC requirements are not yet waived for your account. If we already approved your KYC process and so they should be, then please contact us so we can rectify the issue and update your account accordingly.
  * In case you don't do KYC on your site, then you will need to send us documents that we can use to verify the sender's identity, for more details on this please see the [API reference documentation](https://api.transferzero.com/documentation#documents).

> * If you do not wish to use Cashplus or MAD:Cash as corridor, then we can enable WTR2 rules for you to identify the `sender`.

Reviewer 3: We should start with something like: `We also support sender validation based on WTR2 rules`, then the details on what WTR2 is should be indented into these - so it's a separate block. Notes like cashplus not working should be in this indent. Also make sure to format `MAD::Cash` appropriately, and use block characters

> * If you do not wish to use Cashplus or MAD:Cash as corridor, then we can enable WTR2 rules for you to identify the `sender`.

Reviewer 3: Also we should have a quick note that in case WTR2 is not enabled you have to provide all of the following:

- city, street, postal_code
- birth_date
- email
- phone_number

just to make sure we have a comparison between our normal and our WTR2 operation

> * If you do not wish to use Cashplus or MAD:Cash as corridor, then we can enable WTR2 rules for you to identificate the `sender`.

Reviewer 5: ```suggestion
* If you do not wish to use Cashplus or MAD:Cash as corridor, then we can enable WTR2 rules for you to identify the `sender`.
```

> * If WTR2 is enabled, all you need in order to identificate a `sender` is to provide one of the set of documentation fields defined below.

Reviewer 5: ```suggestion
* If WTR2 is enabled, all you need in order to identify a `sender` is to provide one of the set of documentation fields defined below.
```

* Identification number: Identification number of document used with a 4-character minimum length.

> * Identification type:

Reviewer 3: Make sure this is indented as separate layer. Once creating the commit you can check how it will look like in GitHub by pressing the preview button

  * If you already do KYC on your system, then please contact us and we can waive this requirement from you. In this case you should send us an empty list of documents: `"documents": [ ]` in the request. All of the senders you create in the system will be immediately set to the `approved` state and you won't need to wait for them to get approved.
  * If when creating senders or transactions you get the following error in the response: `"errors":{"documents":[{"error":"blank"}]}` it means that KYC requirements are not yet waived for your account. If we already approved your KYC process and so they should be, then please contact us so we can rectify the issue and update your account accordingly.
  * In case you don't do KYC on your site, then you will need to send us documents that we can use to verify the sender's identity, for more details on this please see the [API reference documentation](https://api.transferzero.com/documentation#documents).

> * WTR2 (Wire Transfer Regulations) was put into place to tackle financial crime and fraudulent activities

Reviewer 3: This is a bit misleading, and we don't really need to tell what WTR2 is. (Especially as wtr2 rules are less strict than our default, so we are much better in tackling financial come then WTR2) What we need to specify that if you wish validation rules for senders can be much more relaxed and based on wtr2 - but if we enable this relaxation then cashplus is out (as it needs a stricter rules than wtr2)

  * If when creating senders or transactions you get the following error in the response: `"errors":{"documents":[{"error":"blank"}]}` it means that KYC requirements are not yet waived for your account. If we already approved your KYC process and so they should be, then please contact us so we can rectify the issue and update your account accordingly.
  * In case you don't do KYC on your site, then you will need to send us documents that we can use to verify the sender's identity, for more details on this please see the [API reference documentation](https://api.transferzero.com/documentation#documents).
* We also support sender validation based on WTR2 rules:
  * If you do not wish to use `Cashplus` or `MAD::Cash` as corridor, then we can enable WTR2 rules for you to identify the `sender`.
  * If WTR2 is enabled, all you need in order to identify a `sender` is to provide one of the set of documentation fields defined below.

>   * Identification number: Identification number of document used with a 4-character minimum length.

Reviewer 3: This 3 should be idented even more, add they are a separate subsection

Author: formatted differently, now should be easier to get the difference between validating a sender through WTR2 or our own traditional requirements

  * Identification type:
    - `DL`: Driving License
    - `PP`: International Passport
    - `ID`: National ID

>     - `OT`: Other

Reviewer 3: The third one (address) is missing

Author: added examples for address and birthdate
